unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PerformanceMeterController/PerformanceMeterController.cs

```diff
@@ -0,0 +1,284 @@
+using System.Collections.Generic;
+using UnityEngine;
+using DCL.FPSDisplay;
+using Newtonsoft.Json;
+
+namespace DCL
+{
+    /// <summary>
+    /// Performance Meter Tool
+    ///
+    /// It samples frames performance data for the target duration and prints a complete report when finished.
+    ///
+    /// There are 2 ways to trigger this tool usage:
+    /// A: While the client is running in the browser, open the browser console and run "clientDebug.RunPerformanceMeterTool(10000);" to run for 10 seconds.
+    /// B: In Unity Editor select the "Main" gameobject and right-click on its DebugBridge Monobehaviour, from there a debug method can be selected to run the tool for 10 seconds.
+    /// </summary>
+    public class PerformanceMeterController
+    {
+        private class SampleData
+        {
+            public int frameNumber;
+            public float millisecondsConsumed;
+            public bool isHiccup = false;
+            public float currentTime;
+            public float fpsAtThisFrameInTime;
+
+            public override string ToString()
+            {
+                return "frame number: " + frameNumber
+                                        + "\n frame consumed milliseconds: " + millisecondsConsumed
+                                        + "\n is hiccup: " + isHiccup
+                                        + "\n fps at this frame: " + fpsAtThisFrameInTime;
+            }
+        }
+
+        private class SamplesFPSComparer : IComparer<SampleData>
```

Comment:
I think we can avoid this whole class just by implementing the `IComparable` interface in `SampleData`.

Reply:
I tried it but couldn't make it work, I'll ask you in a PM

Reply:
I had miss-red `IComparable` as `IComparer` XD, I just implemented your suggestion and it's much better, thanks!